src/Microsoft.Azure.SignalR/Utilities/ServiceRouteHelper.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Connections;
+using Microsoft.Azure.SignalR.Common;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class ServiceRouteHelper
+    {
+        public static async Task RedirectToService(HttpContext context, string hubName, IList<IAuthorizeData> authorizationData)
+        {
+            var handler = context.RequestServices.GetRequiredService<NegotiateHandler>();
+            var loggerFactory = context.RequestServices.GetService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger<ServiceRouteHelper>();
+
+#if !NETCOREAPP3_0
```

Comment:
You can remove this ifdef. Just check for null or empty authorizationData, it's a bit cleaner that way